src/test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceTest.java

```diff
@@ -124,12 +126,110 @@ public void testCreateCatalogObject() {
+        catalogObjectService.updateObjectMetadata("wrong-bucket",
+                                                  NAME,
+                                                  Optional.of("some-kind"),
+                                                  Optional.of("some-contentType"));
+    }
+
+    @Test(expected = CatalogObjectNotFoundException.class)
+    public void testUpdateObjectMetadataWithInvalidObjectName() {
+        BucketEntity bucketEntity = new BucketEntity("bucket", "toto");
+        when(bucketRepository.findOneByBucketName(anyString())).thenReturn(bucketEntity);
+        when(catalogObjectRevisionRepository.findDefaultCatalogObjectByNameInBucket(anyString(),
+                                                                                    anyString())).thenReturn(null);
+        catalogObjectService.updateObjectMetadata(bucketEntity.getBucketName(),
+                                                  "wrong-name",
+                                                  Optional.of("some-kind"),
+                                                  Optional.of("some-contentType"));
+    }
+
     @Test(expected = BucketNotFoundException.class)
     public void testGetCatalogObjectWithInvalidBucket() {
         when(bucketRepository.findOneByBucketName(anyString())).thenReturn(null);
         catalogObjectService.listCatalogObjects("wrong-bucket");
     }
 
+    @Test
+    public void testUpdateObjectMetadata() {
+        long now = System.currentTimeMillis();
+        BucketEntity bucketEntity = new BucketEntity("bucket", "toto");
+        CatalogObjectRevisionEntity catalogObjectEntity = newCatalogObjectRevisionEntity(bucketEntity, now);
+        when(bucketRepository.findOneByBucketName(anyString())).thenReturn(bucketEntity);
+        when(catalogObjectRevisionRepository.findDefaultCatalogObjectByNameInBucket(anyString(),
+                                                                                    anyString())).thenReturn(catalogObjectEntity);
+
+        CatalogObjectMetadata catalogObject = catalogObjectService.updateObjectMetadata(bucketEntity.getBucketName(),
+                                                                                        NAME,
+                                                                                        Optional.of("updated-kind"),
+                                                                                        Optional.of("updated-contentType"));
+
+        assertThat(catalogObject).isNotNull();
+        assertThat(catalogObject.getCommitMessage()).isEqualTo(COMMIT_MESSAGE);
+        assertThat(catalogObject.getContentType()).isEqualTo("updated-contentType");
+        assertThat(catalogObject.getKind()).isEqualTo("updated-kind");
+        assertThat(catalogObject.getName()).isEqualTo(NAME);
+        assertThat(catalogObject.getMetadataList()).isNotEmpty();
+        assertThat(catalogObject.getMetadataList()).hasSize(1);
```

Comment:
You could add a test with the commit time, to ensure that the revision id is good.